integ/src/updater.rs

```diff
@@ -3,149 +3,103 @@
             kube_config_path,
         ])
-        .status()
+        .output()
         .context(update_error::BrupopProcess)?;
 
-    ensure!(brupop_resource_status.success(), update_error::BrupopRun);
-
-    Ok(())
-}
-
-// destroy all brupop resources which were created when integration test installed brupop
-pub async fn delete_brupop_cluster_resources(kube_config_path: &str) -> UpdaterResult<()> {
-    // delete namespaces brupop-bottlerocket-aws. This can clean all resources under this namespace like daemonsets.apps
-    let namespace_deletion_status = Command::new("kubectl")
-        .args([
-            "delete",
-            "namespaces",
-            NAMESPACE,
-            "--kubeconfig",
-            kube_config_path,
-        ])
-        .status()
-        .context(update_error::BrupopCleanUp {
-            cluster_resource: "namespaces",
-        })?;
-    ensure!(namespace_deletion_status.success(), update_error::BrupopRun);
-
-    // delete clusterrolebinding.rbac.authorization.k8s.io
-    for cluster_role_binding in BRUPOP_CLUSTER_ROLE_BINDINGS.iter() {
-        let clusterrolebinding_deletion_status = Command::new("kubectl")
-            .args([
-                "delete",
-                "clusterrolebinding.rbac.authorization.k8s.io",
-                cluster_role_binding,
-                "--kubeconfig",
-                kube_config_path,
-            ])
-            .status()
-            .context(update_error::BrupopCleanUp {
-                cluster_resource: "clusterrolebinding.rbac.authorization.k8s.io",
-            })?;
-        ensure!(
-            clusterrolebinding_deletion_status.success(),
-            update_error::BrupopRun
-        );
-    }
-
-    // delete clusterrole.rbac.authorization.k8s.io
-    for cluster_role in BRUPOP_CLUSTER_ROLES.iter() {
-        let clusterrole_deletion_status = Command::new("kubectl")
-            .args([
-                "delete",
-                "clusterrole.rbac.authorization.k8s.io",
-                cluster_role,
-                "--kubeconfig",
-                kube_config_path,
-            ])
-            .status()
-            .context(update_error::BrupopCleanUp {
-                cluster_resource: "clusterrole.rbac.authorization.k8s.io",
-            })?;
-        ensure!(
-            clusterrole_deletion_status.success(),
-            update_error::BrupopRun
-        );
+    if brupop_resource_output.status.success() {
+        return Ok(());
+    } else {
+        // while creating brupop resources it's possible that test pods already exist on the eks cluster;
+        //therefore, skips test pods creation if those pods already exist.
+        // This wouldn't affect pods deletion.
+        let error_content = String::from_utf8_lossy(&brupop_resource_output.stderr).to_string();
+        match error_content.contains("AlreadyExists") {
+            true => {
+                info!("brupop resources have been created, skip...");
+                return Ok(());
```

Comment:
Is it possible for us to check that those resources are using the same code as what we're currently trying to test? That might be useful info to someone running tests.

Reply:
"the same code" here you mean is `bottlerocekt-update-operator.yaml`?  For example, developers are going to test some new changes on brupop resources, but due to the old brupop resources had been created, so when run a new integration test, it will skip the new brupop resources installing. Is my understand correct here or ? : )

Reply:
Yes! That's what I was thinking of.

Reply:
I chatted with Tianhao about the specifics of this: basically test resources should always be cleaned between runs, and shared resources are otherwise acceptable if testing for multiple architectures. I'm okay with proceeding with this implementation.